Remove VpnServiceStateLogger and move that logic to VPN service

aitorvs opened the pull request:
Task/Issue URL: https://app.asana.com/0/488551667048375/1203665161728424/f

### Description
See [Asana task](https://app.asana.com/0/488551667048375/1203665161728424/f) for more context

### Steps to test this PR
_Fresh install_
- [x] fresh install from this branch
- [x] enable AppTP
- [x] verify AppTP is enabled after onboarding
- [x] disable AppTP
- [x] verify AppTP is disabled
- [x] re-enable AppTP
- [x] verify AppTP is enabled
- [x] kill the VPN process using eg. AS
- [x] verify AppTP is re-enabled

_App update_
- [x] install app from develop and enable AppTP
- [x] update from this branch
- [x] verify AppTP re-enables automatically at launch

_Kill VPN process_
- [x] install from this branch and enable AppTP
- [x] with AS, kill the .vpn process
- [x] verify the VPN is re-enabled automatically

_VPN revoke_
- [x] install from this branch
- [x] install another VPN app
- [x] enable AppTP
- [x] enable the 3rd party VPN app
- [x] verify AppTP is disable and in the AppTP screen the yellow banner warns that another VPN may have disabled AppTP
- [x] re-enable AppTP
- [x] verity AppTP is re-enabled

_AppTP toggle flicker_
- [x] fresh install from develop
- [x] enable apptP (go through onboarding)
- [x] verify the toggle goes from disabled -> enabled -> disabled -> enabled
- [x] fresh install from this branch
- [x] enable apptP (go through onboarding)
- [x] verify the toggle goes to (just) from disabled -> enabled

aitorvs commented:
Current dependencies on/for this PR:
* develop
  * **PR #2698** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2698" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2698?utm_source=stack-comment).

karlenDimla commented on app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt:
```diff
@@ -95,10 +104,14 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
 
     @Inject lateinit var vpnServiceStateStatsDao: VpnServiceStateStatsDao
 
+    @Inject lateinit var vpnDatabase: VpnDatabase
```
I don't think we need to inject this since we are already injecting vpnServiceStateStatsDao above.

aitorvs replied:
goot point. Done

karlenDimla commented on app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt:
```diff
@@ -545,6 +569,64 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
         )
     }
 
+    private suspend fun monitorVpnAlwaysOnState() = withContext(Executors.newSingleThreadExecutor().asCoroutineDispatcher()) {
```
What you think about keeping the VpnServiceStateLogger class and keeping all state related updates there. We can have something like:

interface VpnServiceStateLogger {
     fun logEnabled()
     fun logDisabled()
     fun logEnabling()
     fun monitorVpnAlwaysOnState()
     fun getCurrentState():  VpnServiceState
}


I think TrackerBlockVpnService is starting to be big so it might be a good idea to not keep putting more logic here unless it is really necessary.

aitorvs replied:
Not sure what you mean, do you mean just extracting that logic to a separate class but it not being a VPN callbacks plugin?

I agree VPN service is already somewhat big. Not sure about extracting that for now. I'd rather approach if/when extracting logic from the service more holistically (?)

karlenDimla replied:
Yeah just extracting the logic to a separate class that we call in TrackerBlockingVpnService instead.

What do you mean by holistically? 

This is just a NIT anyway as I wanted that the service is big and hard to understand already.

aitorvs replied:
What I meant is having a dedicated task and there's probably more things we should/need to extract. So yeah, I am basically agreeing with you but wanted to have dedicated time for it, if that make sense.

aitorvs commented:
@karlenDimla, added https://github.com/duckduckgo/Android/pull/2698/commits/74798caa9b0bd4994f5903066bcd87b658eccd03 addressing some of your comments and fixing a toggle flickering we had since we added the ENABLING state.
